src/k8s/pkg/k8sd/app/hooks_bootstrap.go

```diff
@@ -50,10 +50,13 @@ func onBootstrapWorkerNode(s *state.State, encodedToken string) error {
 	}
 
 	// TODO(neoaggelos): figure out how to use the microcluster client instead
-
+	timeout := 30 * time.Second
+	if deadline, set := s.Context.Deadline(); set {
```

Comment:
Is `s.Context.Deadline()` ever set on this call? If not we will again use a fixed timeout. Should we adjust this to handle a cancelled context like we've done yesterday?